From f8b37c11b6736cf404e3c68d32eefc98f251b9b6 Mon Sep 17 00:00:00 2001
From: Daniel Mueller <deso@posteo.net>
Date: Sat, 8 Dec 2018 09:08:18 -0800
Subject: [PATCH] End attributes with a trailing comma

With changes leading up to 1.0.0, rustfmt removes the trailing comma
from attribute lists such as:
  > #![deny(
  >   unused_results,
  >   warnings,
  > )]

This behavior is undesired, as it causes unnecessary churn in the diff
when adding a new element to the end of the attribute list.
With this patch we bring back the original behavior of keeping (or
adding) the trailing comma.
---
 src/tools/rustfmt/src/overflow.rs | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/overflow.rs b/src/overflow.rs
index 4a583f..fb6b2a 100644
--- a/src/tools/rustfmt/src/overflow.rs
+++ b/src/tools/rustfmt/src/overflow.rs
@@ -611,6 +611,12 @@ impl<'a> Context<'a> {
             _ => false,
         };
 
+        let trailing_separator = if self.context.use_block_indent() && list_items.len() > 1 && ends_with_newline {
+          SeparatorTactic::Always
+        } else {
+          trailing_separator
+        };
+
         let fmt = ListFormatting::new(self.nested_shape, self.context.config)
             .tactic(tactic)
             .trailing_separator(trailing_separator)
-- 
2.19.2

